### PR TITLE
rpc: Cleanup CacheBlockMetaService

### PR DESCRIPTION
**Pull request by steviez**
#### Summary of Changes
- Add thread start/stop logs so we can track lifetime
- Bubble up errors from writing Blockstore - stop the validator instead of just writing log
    - Failure to write could indicate missing data, better to fail loudly
    
I though about renaming `CacheBlockMeta*` to `BlockMeta*` as part of this but held off - I'm acting upon the assumption that my reviewers don't mind more small PR's over fewer large ones

### CODE REVIEW

File: rpc/src/cache_block_meta_service.rs [hunk 1]
```diff
@@ -30,40 +30,39 @@ impl CacheBlockMetaService {
...
-    fn cache_block_meta(bank: &Bank, blockstore: &Blockstore) {
-        if let Err(e) = blockstore.cache_block_time(bank.slot(), bank.clock().unix_timestamp) {
-            error!("cache_block_time failed: slot {:?} {:?}", bank.slot(), e);
-        }
-        if let Err(e) = blockstore.cache_block_height(bank.slot(), bank.block_height()) {
-            error!("cache_block_height failed: slot {:?} {:?}", bank.slot(), e);
-        }
+    fn cache_block_meta(bank: &Bank, blockstore: &Blockstore) -> Result<(), BlockstoreError> {
+        blockstore.cache_block_time(bank.slot(), bank.clock().unix_timestamp)?;
+        blockstore.cache_block_height(bank.slot(), bank.block_height())
```
Comment (bw-solana) [inline rpc/src/cache_block_meta_service.rs:65]: does it matter that we're losing observability for which insertion (block time vs. block height) failed?

File: rpc/src/cache_block_meta_service.rs [hunk 2]
```diff
@@ -30,40 +30,39 @@ impl CacheBlockMetaService {
...
+                info!("CacheBlockMetaService has started");
+                loop {
+                    if exit.load(Ordering::Relaxed) {
                         break;
                     }
-                    Ok(bank) => {
-                        let mut cache_block_meta_timer = Measure::start("cache_block_meta_timer");
-                        Self::cache_block_meta(&bank, &blockstore);
-                        cache_block_meta_timer.stop();
-                        if cache_block_meta_timer.as_ms() > CACHE_BLOCK_TIME_WARNING_MS {
```
Comment (bw-solana) [inline rpc/src/cache_block_meta_service.rs:46]: what's the idea behind getting rid of this?

**Review by bw-solana — COMMENTED** [review]
Adding logging and bubbling error seem like positive additions.

Asked a couple questions about the things we're losing.

File: rpc/src/cache_block_meta_service.rs (hunk 2 above)
Comment (steviez) [reply rpc/src/cache_block_meta_service.rs:46]: This timer seemed like overkill for this service. If it takes more than 150ms to push < 50 bytes into the Blockstore, the system is going to fail in much more spectacular ways

File: rpc/src/cache_block_meta_service.rs (hunk 1 above)
Comment (steviez) [reply rpc/src/cache_block_meta_service.rs:65]: My opinion is no; there is nothing really special happening in either of these columns. More so, rocksdb errors are typically "sticky". So, if the first one fails, I expect the second one (and all other writes) to fail. So, while this thread will bail out, I imagine `ReplayStage` or `WindowService` will have panicked somewhere

Comment (bw-solana) [reply rpc/src/cache_block_meta_service.rs:65]: yeah, that's my understanding as well

**Review by bw-solana — APPROVED** [review]
LGTM

**Pull request merged**